src/sage/categories/modules_with_basis.py

```diff
@@ -729,7 +729,7 @@ def echelon_form(self, elements, row_reduced=False, order=None):
 
         def submodule(self, gens, check=True, already_echelonized=False,
                       unitriangular=False, support_order=None, category=None,
-                      *args, **opts):
+                      parent_class=None, *args, **opts):
```

Comment:
Perhaps that should be `submodule_class` (like existing `ideal_class` parameters)?